src/framegrab/grabber.py

```diff
@@ -666,9 +669,44 @@ def __init__(self, config: dict):
         thread.daemon = True
         thread.start()
 
+    def _substitute_rtsp_password(self, config: dict) -> dict:
+        """
+        Substitutes the password placeholder in the rtsp_url with the actual password
+        from an environment variable.
+        The URL format should take this format
+            Ex: rtsp://admin:{{RTSP_PASSWORD}}@10.0.0.0/cam/realmonitor?channel=1&subtype=0
+
+        This function looks for a string between {{ and }} to find an environment variable with that name.
```

Comment:
```suggestion
        This function looks for an all uppercase name between {{ and }} to find an environment variable with that name.
```

Your regex is in fact looking for an arbitrary string, but that's not good.  I don't know if environment variables literally can have spaces or punctuation, but they definitely shouldn't.  And they're almost always uppercase, so I think it's better to restrict to those.